### ISSUE TITLE
APIlogin handler is fired AFTER registerEventHooks so hooks register in a logged out session

### ISSUE THREAD

**Issue by mapkyca** [idno/Known#811]
Tried passing syndication[]=twitter and syndication[]=twitter::mapkyca, neither worked.

**Comment by mapkyca** [comment]
So, I watched a POST made with the ux, and I've sent the identical info via the api, and it doesn't work. Not sure what's going on, but there seems to be a disconnect with what the UX sends and what the Status plugin expects.

So, when I send via API syndication[]=twitter::mapkyca, which is what the UX sends (url encoded of course), I get the following error in the log:

```
[Sun Mar 22 10:08:21 2015] [error] [client xx.xx.xx.xx] SERVICES for note are ["facebook","twitter","linkedin","buffer","appnet","known","github"]
[Sun Mar 22 10:08:21 2015] [error] [client xx.xx.xx.xx] SELECTED SERVICES ARE ["twitter::mapkyca"]
```

Which implies that twitter expects the old form posting (this despite the fact that it works successfully in the UX).

When I send just syndication[]=twitter (the old form), I get:

```
[Sun Mar 22 10:08:08 2015] [error] [client xx.xx.xx.xx] SERVICES for note are ["facebook","twitter","linkedin","buffer","appnet","known","github"]
[Sun Mar 22 10:08:08 2015] [error] [client xx.xx.xx.xx] SELECTED SERVICES ARE ["twitter"]
[Sun Mar 22 10:08:08 2015] [error] [client xx.xx.xx.xx] Known (mapkyca.com): INFO - testing something: 17 characters 4
[Sun Mar 22 10:08:08 2015] [error] [client xx.xx.xx.xx] Known (mapkyca.com): INFO - Nothing was posted to Twitter: stdClass::__set_state(array(\n   'errors' => \n  array (\n    0 => \n    stdClass::__set_state(array(\n       'code' => 220,\n       'message' => 'Your credentials do not allow access to this resource.',\n    )),\n  ),\n)) 4
```

Which implies that it got further, but the token was invalid. This despite the fact that it frickin' works using the UX poster.

I've tried reconnecting the account with no joy. Any thoughts?

**Comment by mapkyca** [comment]
I'm wondering if it's a session problem, possibly. Although state is being preserved via cookie...

**Comment by mapkyca** [comment]
Interesting... 

So it *looks* like the posse event handler isn't being triggered, because 'hasTwitter()' is returning false on the POST, but returns true the second time around on the GET for the object.

I'm wondering whether the registerEventHooks() code is being triggered *before* the login via API code, so that the environment is set up before a user is seen to be logged in (but the canEdit()) is of course being hit after login has occurred.

So, basically, it looks like the API handling code needs to happen a bit earlier on in the stack.

**Comment by mapkyca** [comment]
Confirmed. 

Registration of events happens before APILogin(), so occurs in a logged out environment.
